cmd/main.go

```diff
@@ -66,22 +69,25 @@ func handle() {
 
 	gceDriver := driver.GetGCEDriver()
 
-	//Initialize GCE Driver (Move setup to main?)
-	cloudProvider, err := gce.CreateCloudProvider(vendorVersion, *gceConfigFilePath)
+	// Initialize GCE Driver (Move setup to main?)
+	cloudProvider, err := gce.CreateCloudProvider(vendorVersion, *cloudConfigFilePath)
 	if err != nil {
 		klog.Fatalf("Failed to get cloud provider: %v", err)
 	}
 
 	mounter := mountmanager.NewSafeMounter()
 	deviceUtils := mountmanager.NewDeviceUtils()
 	statter := mountmanager.NewStatter()
-	ms, err := metadataservice.NewMetadataService()
-	if err != nil {
-		klog.Fatalf("Failed to set up metadata service: %v", err)
+
+	var ms metadataservice.MetadataService
+	if !*runNodeService {
```

Comment:
i thought we *do* set up the metadata service if we're running nodeService?

Reply:
we actually need this in both node and controller service.

Reply:
Do you want me to add another flag indicating the the driver is not running on a GCE instance? The metadata service is only available there.

Reply:
lets call it `use-metadata-server` and do some validation that `if !use-metadataserver then project & zone must be set on cloud config`. Basically the metadataserver becomes the "backup" method to get zone and project on a GCE instance if the project and zone are not provided directly